radar-upload-backend/src/main/java/org/radarbase/upload/resource/RecordResource.kt

```diff
@@ -152,11 +150,60 @@ class RecordResource {
         return recordMapper.fromRecords(recordRepository.poll(imposedLimit), imposedLimit)
     }
 
+    @GET
+    @Path("{recordId}/metadata")
+    fun getRecordMetaData(@PathParam("recordId") recordId: Long): RecordMetadataDTO {
+
+        val record = recordRepository.read(recordId)
+                ?: throw NotFoundException("Record with ID $recordId does not exist")
+
+        return recordMapper.fromMetadata(record.metadata)
+    }
+
+    @POST
+    @Path("{recordId}/metadata")
+    fun updateRecordMetaData(metaData: RecordMetadataDTO, @PathParam("recordId") recordId: Long): RecordMetadataDTO {
+
+        val record = recordRepository.read(recordId)
+                ?: throw NotFoundException("Record with ID $recordId does not exist")
+
+        record.metadata.canBeUpdatedTo(metaData)
+
+        val updatedRecord = recordRepository.update(recordMapper.toMetadata(metaData, record.metadata))
+
+        return recordMapper.fromMetadata(updatedRecord)
+    }
+
+    private fun RecordMetadata.canBeUpdatedTo(metaDataToUpdate: RecordMetadataDTO) {
+
+        if (this.revision != metaDataToUpdate.revision)
+            throw BadRequestException("Requested meta data revision ${metaDataToUpdate.revision} " +
+                    "should match the latest revision stored ${this.revision}")
+
+        if (metaDataToUpdate.status == RecordStatus.PROCESSING.toString()
+                && this.status != RecordStatus.QUEUED) {
+            throw ConflictException("Record cannot be updated: Conflict in record meta-data status. " +
+                    "Found ${this.status}, expected ${RecordStatus.QUEUED}")
+        }
+
+        if (metaDataToUpdate.status == RecordStatus.SUCCEEDED.toString()
+                || metaDataToUpdate.status == RecordStatus.FAILED.toString()
+                && this.status != RecordStatus.PROCESSING) {
+            throw ConflictException("Record cannot be updated: Conflict in record meta-data status. " +
+                    "Found ${this.status}, expected ${RecordStatus.QUEUED}")
+        }
+
+    }
+
+    @GET
+    @Path("{recordId}/logs/")
+    fun getRecordLogs(@PathParam("recordId") recordId: Long): LogsDto {
 
-    // TODO: get metadata path
-    // TODO: update metadata path
+        val record = recordRepository.read(recordId)
+                ?: throw NotFoundException("Record with ID $recordId does not exist")
 
+        return recordMapper.fromMetadataToLogs(record.metadata)
```

Comment:
I would propose to just return the raw logs in this call, as plain text for now. We can allow additional content types later on if we modify the database schema to store the content type of the logs. Even then, we probably want to limit the types of data logs can take, e.g. xml, text or JSON.